### PR TITLE
Simplify Inference

### PR DESCRIPTION
**Pull request by l-k-11235**
This PR aims to simplify inference by adding a `translate_strings` method to be applied to a list of raw strings.
It is used and it has been tested in `eval_llm/MMLU/run_mmlu_opennmt.py` and in  `docs/source/examples/replicate_vicuna/chatbot.py`.

### CODE REVIEW

File: onmt/translate/translator.py [hunk 1]
```diff
@@ -298,6 +304,34 @@ def _gold_score(
...
+
+        infer_iter = IterOnDevice(infer_iter, self._gpu)
+        _, preds = self._translate(
+            infer_iter,
+            transform=infer_iter.transform,
+            attn_debug=self.infer_opt.attn_debug,
+            align_debug=self.infer_opt.align_debug,
+        )
+
+        output_sentences = [
```
Comment (vince62s) [inline onmt/translate/translator.py:329]: I don't think this part should be included in this method. it was specific to the MMLU evaluation

Comment (l-k-11235) [reply onmt/translate/translator.py:331]: I passed `attn_debug` and `align_debug` as arguments with the default value False.

**Comment by l-k-11235** [comment]
I should add an example for the inference from a list of strings  here https://opennmt.net/OpenNMT-py/quickstart.html#step-3-translate

**Comment by vince62s** [comment]
also remove the infer_opt changes in translator.py

File: eval_llm/MMLU/run_mmlu_opennmt.py [hunk 2]
```diff
@@ -180,21 +175,8 @@ def evaluate(opt):
...
-            infer_iter,
-            transform=infer_iter.transform,
-            attn_debug=opt.attn_debug,
-            align_debug=opt.align_debug,
-        )
-        pred_answers = [
-            x.lstrip() for sublist in preds for x in sublist
-        ]  # flatten the list of list
+        pred_answers = translate(opt, translator, src)
+        pred_answers = [x.strip() for x in pred_answers]
```
Comment (vince62s) [inline eval_llm/MMLU/run_mmlu_opennmt.py:179]: only "lstrip()" to remove the leading space

**Pull request closed without merging**